src/composables/trade/useSor.ts

```diff
@@ -402,8 +402,12 @@ export default function useSor({
       if (!sorReturn.value.hasSwaps) {
         priceImpact.value = 0;
       } else {
-        tokenInAmount = await adjustedPiAmount(tokenInAmount, tokenOutAddress);
-
+        if (isMainnet.value) {
+          tokenInAmount = await adjustedPiAmount(
+            tokenInAmount,
+            tokenOutAddress
+          );
```

Comment:
Are you sure we don't actually need to do this on L2s like Polygon?

Reply:
I am not sure, @markusbkoch can you help here please

Reply:
I don't think this conditional should be introduced here. I don't think it would break anything under the current circumstances, but it seems to me like `adjustedPiAmount` is meant to be a catch-all for cases like the stETH/wstETH one, but not limited to this specific case.

Reply:
I changed the condition to `isMainnet.value || isArbitrum.value` as there are currently rate providers addresses only for these networks

Reply:
Right, what I mean is that IIUC this check is redundant with the check that already exists within the `adjustedPiAmount` function. So unless I'm missing something this condition means we have to maintain two pieces of code instead of one when a new rate provider for wstETH comes along on Polygon for example, or when we add support for wrapping a new token that's similar to stETH. Like I said, the conditional here doesn't seem to break anything for the current state of things. But it does seem to add overhead to our work in the future.

Reply:
if we remove condition, it will break calculation only on goerli, can we add rate provider there?

Reply:
thanks, I will remove conditions